reports_automation/report_create.py

```diff
@@ -0,0 +1,118 @@
+import requests
+import os, json,sys
+from configparser import ConfigParser,ExtendedInterpolation
+
+sys.path.insert(0, '/opt/sparkjobs/ml-analytics-service/reports_automation/')
+from mongo_logging import *
+
+# Read the Config
+config_path = os.path.split(os.path.dirname(os.path.abspath(__file__)))
+config = ConfigParser(interpolation=ExtendedInterpolation())
+config.read(config_path[0] + "/report_config.ini")
+
+# Required field gathering for API
+base_url = config.get("API_ENDPOINTS","base_url")
+headers_api = {
+        'Content-Type': config.get("API_HEADERS", "content_type"),
+        'Authorization' : config.get("API_HEADERS","authorization_access_token")
+    }
+
+folder_path = config.get("REPORTS_FILEPATH","base_path") + config.get("REPORTS_FILEPATH","folder_name") + "/"
+
+# Creation of chart using Json config making an API call
+def backEnd_create(file_name):
+     try :
+        url_backend_create = base_url + config.get("API_ENDPOINTS","backend_create")
+        file_path = folder_path + "backEndConfig" + "/" + file_name
+        with open(file_path) as data_file:
+            json_config = json.load(data_file)
+            json_config["request"]["createdBy"] = config.get("JSON_VARIABLE","createdBy")
+            json_config["request"]["config"]["container"] = config.get("JSON_VARIABLE","container")
+            json_config["request"]["config"]["store"] = config.get("JSON_VARIABLE","store")
+            json_config["request"]["config"]["key"] = config.get("JSON_VARIABLE","key")
+            json_config["request"]["config"]["reportConfig"]["mergeConfig"]["postContainer"] = config.get("JSON_VARIABLE","container")
+            json_config["request"]["config"]["reportConfig"]["mergeConfig"]["container"] = config.get("JSON_VARIABLE","container")
+
+            doc = {
+                    "configFileName" : file_name,
+                    "config" : json.dumps(json_config),
+                    "operation": "backEnd_create"
+                  }
+
+        #Api call
+        response_api = requests.post(
+                url_backend_create,
+                data= json.dumps(json_config),
+                headers=headers_api
+            )
+
+        # Based on status concluding logging the output
+        if response_api.status_code == 200:
+           response_data = response_api.json()
+           if response_data["params"]["status"] ==  "failed":
+              doc["errmsg"] = "ReportId already Exists"
+              type = "error"
+           else:
+              type = "crud"
+
+        else:
+           doc["errmsg"] = str(response_api.status_code)  + response_api.text
+           type = "error"
+
+     except Exception as exception:
+             doc["errmsg"] = "Exception message {}: {}".format(type(exception).__name__, exception)
+             type = "exception"
+
+     insert_doc(doc,type)
```

Comment:
@Ashwiniev95 Let's log the entire success and error response from these APIs in DB in addition to what we already have.

Reply:
Added